glyphs-reader/src/font.rs

```diff
@@ -1032,18 +1133,71 @@ fn user_to_design_from_axis_location(
     Some(axis_mappings)
 }
 
-fn extract_axis_mappings(from: &RawFont) -> BTreeMap<String, RawUserToDesignMapping> {
+fn add_mapping_if_new(
+    mappings: &mut RawUserToDesignMapping,
+    user: OrderedFloat<f32>,
+    design: OrderedFloat<f32>,
+) {
+    if mappings.iter().any(|(u, d)| *u == user || *d == design) {
+        return;
+    }
+    mappings.push((user, design));
+}
+
+/// * <https://github.com/googlefonts/glyphsLib/blob/6f243c1f732ea1092717918d0328f3b5303ffe56/Lib/glyphsLib/builder/axes.py#L128>
+/// * <https://github.com/googlefonts/glyphsLib/blob/6f243c1f732ea1092717918d0328f3b5303ffe56/Lib/glyphsLib/builder/axes.py#L353>
+fn add_mappings_from_instances(
+    instances: &[Instance],
+    mut mappings: BTreeMap<String, RawUserToDesignMapping>,
```

Comment:
Why don’t we take mappings by mutable reference here? Also, the parameter order is different from `add_mapping_if_new` which is confusing.

Reply:
I wanted to be able to write `add_mappings(blah)` in return position. In the light of a new day i don't think it's a win, will change to mut ref.

Reply:
...actually to it's own type